packages/cli/commands/project/cloneApp.js

```diff
@@ -0,0 +1,177 @@
+    logger.log(
+      i18n(`${i18nKey}.errors.invalidAccountTypeDescription`, {
+        useCommand: uiCommandReference('hs accounts use'),
+        authCommand: uiCommandReference('hs auth'),
+      })
+    );
+    uiLine();
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  let appId;
+  let appName;
+  let location;
+  try {
+    const appIdResponse =
+      'appId' in options
+        ? { appId: options.appId }
+        : await selectPublicAppPrompt({
+            accountId,
+            accountName,
+            options,
+            migrateApp: false,
+          });
+    appId = appIdResponse.appId;
+
+    const selectedApp = await fetchPublicAppMetadata(appId, accountId);
+    appName = selectedApp.name;
+
+    const locationResponse =
+      'location' in options
+        ? { location: options.location }
+        : await promptUser({
+            name: 'location',
+            message: i18n(`${i18nKey}.enterLocation`),
+            default: path.resolve(getCwd(), appName),
+            validate: input => {
+              if (!input) {
+                return i18n(`${i18nKey}.errors.locationRequired`);
+              }
+              if (fs.existsSync(input)) {
+                return i18n(`${i18nKey}.errors.invalidLocation`);
+              }
+              if (!isValidPath(input)) {
+                return i18n(`${i18nKey}.errors.invalidCharacters`);
+              }
+              return true;
+            },
+          });
+    location = locationResponse.location;
+  } catch (error) {
+    logApiErrorInstance(error, new ApiErrorContext({ accountId }));
+    process.exit(EXIT_CODES.ERROR);
+  }
+  try {
+    SpinniesManager.init();
+
+    SpinniesManager.add('cloneApp', {
+      text: i18n(`${i18nKey}.cloneStatus.inProgress`),
+    });
+
+    const { exportId } = await cloneApp(accountId, appId);
+    const { status } = await poll(checkCloneStatus, accountId, exportId);
+    if (status === 'SUCCESS') {
+      const absoluteDestPath = path.resolve(getCwd(), location);
+      const zippedProject = await downloadClonedProject(accountId, exportId);
+
+      await extractZipArchive(
+        zippedProject,
+        appName,
+        path.resolve(absoluteDestPath),
+        { includesRootDir: true, hideLogs: true }
+      );
+
+      SpinniesManager.succeed('cloneApp', {
+        text: i18n(`${i18nKey}.cloneStatus.done`),
+        succeedColor: 'white',
+      });
+      logger.log('');
+      uiLine();
+      logger.success(i18n(`${i18nKey}.cloneStatus.success`, { location }));
+      logger.log('');
+      process.exit(EXIT_CODES.SUCCESS);
+    }
+  } catch (error) {
+    SpinniesManager.fail('cloneApp', {
+      text: i18n(`${i18nKey}.cloneStatus.failure`),
+      failColor: 'white',
+    });
+    // Migrations endpoints return a response object with an errors property. The errors property contains an array of errors.
+    if (error.errors) {
+      error.errors.forEach(logApiErrorInstance);
```

Comment:
Should these include an `ApiErrorContext` with `accountId` as well?

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1085/commits/247b941649560ed561734257bca012ddf2d65f02